definitions/npm/detox_v7.0.x/test_detox_v7.0.x.js

```diff
@@ -0,0 +1,170 @@
+// ------------------
+ element(by.id('testElement')).tap()
+ element(by.id('testElement')).longPress()
+
+// $ExpectError
+ element(by.id('testElement')).multiTap(true)
+ element(by.id('testElement')).multiTap(3)
+
+// $ExpectError
+ element(by.id('testElement')).tapAtPoint()
+ element(by.id('testElement')).tapAtPoint({x: 5, y: 10})
+
+// $ExpectError
+ element(by.id('testElement')).typeText(false)
+ element(by.id('testElement')).typeText('type this')
+
+// $ExpectError
+ element(by.id('testElement')).replaceText()
+ element(by.id('testElement')).replaceText('text')
+
+ element(by.id('testElement')).clearText()
+
+// $ExpectError
+ element(by.id('testElement')).scroll()
+ element(by.id('testElement')).scroll(123, 'down')
+
+// $ExpectError
+ element(by.id('testElement')).scrollTo('down')
+ element(by.id('testElement')).scrollTo('left')
+
+// $ExpectError
+ element(by.id('testElement')).swipe('sideways', 'medium', 0.5)
+ element(by.id('testElement')).swipe('down', 'fast', 0.5)
+
+
+// Expect API
+// ----------
+expect(element(by.id('UniqueId204'))).toBeVisible()
+expect(element(by.id('UniqueId204'))).toBeNotVisible()
+expect(element(by.id('UniqueId204'))).toBeVisible()
+expect(element(by.id('UniqueId204'))).toExist()
+// $ExpectError
+expect(element(by.id('UniqueId204'))).toHaveId(123)
+expect(element(by.id('UniqueId204'))).toHaveId('someTestID')
+
+// $ExpectError
+expect(element(by.id('UniqueId204'))).toHaveLabel()
+expect(element(by.id('UniqueId204'))).toHaveLabel('testLabel')
+
+// $ExpectError
+expect(element(by.id('UniqueId204'))).toHaveText(123)
+expect(element(by.id('UniqueId204'))).toHaveText('some text')
+// $ExpectError
+expect(element(by.id('UniqueId204'))).toHaveValue(true)
+expect(element(by.id('UniqueId204'))).toHaveValue("1")
+expect(element(by.id('UniqueId204'))).toNotExist()
+
+// waitFor API
+waitFor(element(by.id('UniqueId204'))).toBeNotVisible()
+waitFor(element(by.id('UniqueId204'))).toBeNotVisible()
+waitFor(element(by.id('UniqueId204'))).toBeVisible()
+// $ExpectError
+waitFor(element(by.id('UniqueId204'))).toExist().something()
+waitFor(element(by.id('UniqueId204'))).toExist()
+waitFor(element(by.id('UniqueId204'))).toHaveText('text')
+waitFor(element(by.id('UniqueId204'))).toHaveValue("0")
+waitFor(element(by.id('UniqueId204'))).toNotExist()
+// $ExpectError
+waitFor(element(by.id('UniqueId204'))).whileElement('someID')
+waitFor(element(by.id('UniqueId204'))).whileElement(by.id('ScrollView630'))
+// $ExpectError
+waitFor(element(by.id('UniqueId204'))).withTimeout(false)
+waitFor(element(by.id('UniqueId204'))).withTimeout(1000)
+
+
+// Example tests
+const e2e = {
+    NATIVE_BUTTON: "NATIVE_BUTTON",
+    WELCOME_VIEW: "WELCOME_VIEW"
+}
+
+const welcome = element(by.id(e2e.WELCOME_VIEW));
+expect(welcome).toBeVisible();
+const welcome_text = "Shake your phone to open the developer menu.";
+expect(element(by.text(welcome_text))).toBeVisible();
+const btn = element(by.id(e2e.NATIVE_BUTTON));
+btn.tap();
```

Comment:
Seems like everything is here. I have a few questions though, written above.

Thanks for taking care of Detox :)